clients/rust/tests/initialize_sol_staker_stake.rs

```diff
@@ -0,0 +1,450 @@
+    let validator_stake_account = ValidatorStake::from_bytes(account.data.as_ref()).unwrap();
+    assert_eq!(
+        validator_stake_account.total_staked_lamports_amount,
+        1_000_000_000
+    );
+}
+
+#[tokio::test]
+async fn fail_initialize_sol_staker_stake_with_initialized_account() {
+    let mut program_test = ProgramTest::new(
+        "paladin_stake_program",
+        paladin_stake_program_client::ID,
+        None,
+    );
+    program_test.add_program(
+        "paladin_sol_stake_view_program",
+        paladin_sol_stake_view_program_client::ID,
+        None,
+    );
+    let mut context = program_test.start_with_context().await;
+
+    // Given a config and a validator stake accounts.
+
+    let config_manager = ConfigManager::new(&mut context).await;
+    let stake_manager = ValidatorStakeManager::new(&mut context, &config_manager.config).await;
+
+    // And we create a SOL stake account.
+
+    let stake_state = Keypair::new();
+    let stake_amount = 1_000_000_000;
+    let withdrawer = context.payer.pubkey();
+
+    create_stake_account(
+        &mut context,
+        &stake_state,
+        &Authorized::auto(&withdrawer),
+        &Lockup::default(),
+        stake_amount,
+    )
+    .await;
+
+    let stake_state = stake_state.pubkey();
+    delegate_stake_account(&mut context, &stake_state, &stake_manager.vote).await;
+
+    // And we initialize the SOL staker stake account.
+
+    let (stake_pda, _) = find_sol_staker_stake_pda(&stake_state, &config_manager.config);
+
+    let transfer_ix = system_instruction::transfer(
+        &context.payer.pubkey(),
+        &stake_pda,
+        context
+            .banks_client
+            .get_rent()
+            .await
+            .unwrap()
+            .minimum_balance(SolStakerStake::LEN),
+    );
+
+    let initialize_ix = InitializeSolStakerStakeBuilder::new()
+        .config(config_manager.config)
+        .sol_staker_stake(stake_pda)
+        .validator_stake(stake_manager.stake)
+        .stake_state(stake_state)
+        .sol_stake_view_program(paladin_sol_stake_view_program_client::ID)
+        .instruction();
+
+    let tx = Transaction::new_signed_with_payer(
+        &[transfer_ix, initialize_ix],
+        Some(&context.payer.pubkey()),
+        &[&context.payer],
+        context.last_blockhash,
+    );
+    context.banks_client.process_transaction(tx).await.unwrap();
+
+    // When we try to initialize the SOL staker stake account again.
+
+    let initialize_ix = InitializeSolStakerStakeBuilder::new()
+        .config(config_manager.config)
+        .sol_staker_stake(stake_pda)
+        .validator_stake(stake_manager.stake)
+        .stake_state(stake_state)
+        .sol_stake_view_program(paladin_sol_stake_view_program_client::ID)
+        .instruction();
+
+    let tx = Transaction::new_signed_with_payer(
+        &[initialize_ix],
+        Some(&context.payer.pubkey()),
+        &[&context.payer],
+        context.last_blockhash,
+    );
```

Comment:
Be careful, this can be flaky, since you're doing the same exact transaction twice, with the same signature. It's better to get a new blockhash before doing this.

Reply:
Good point!